src/Screens/Main/Settings/Password.tsx

```diff
@@ -46,21 +46,34 @@ export default ({ navigation }: Props) => {
     repeatedNewPassword.length > 0 &&
     newPassword === repeatedNewPassword;
 
+  const onButtonPress = () => {
+    dispatch(state.changePassword({ currentPassword, newPassword }));
+  };
+
   const onGoBack = () => {
     navigation.goBack();
   };
 
-  const onButtonPress = () => {
-    dispatch(state.changePassword({ currentPassword, newPassword }));
+  const resetAndGoBack = () => {
+    setCurrentPassword('');
+    setNewPassword('');
+    setRepeatedNewPassword('');
+    dispatch(state.reset);
+    onGoBack();
```

Comment:
Why u need to reset? Does the navi cache this screen or something?

Reply:
Actually did not even try without it. Dont need it.